docs/backend/backend-overview.md

```diff
@@ -0,0 +1,210 @@
+# Backend Service - Future Vision
+
+This document outlines a future high-level vision for what the Backend Service
+could look like as an evolution from where we are today.
+
+Goals:
+
+- Specify key requirements for an evolved high-level architecture
+
+Non Goals:
+
+- Providing a detailed description of how the current backend is implemented.
+- Getting into implementation details on a component level. We may cover key
+  technology choices, but not details on how they are to be used.
+
+## Personas
+
+The following personas will be referenced throughout the document:
+
+- **Cluster Administrator**: The person responsible for managing a Kubernetes cluster.
+  This person cares about all hardware and its associated hardware and
+  cluster-wide software resources.
+
+- **Backend Service Owner**: The person responsible for managing the Backend Service
+  on top of the cluster provided by the Cluster Administrator.
+
+- **InstructLab Model Developer**: This person is a user of the Backend Service.
```

Comment:
Would you bucket a Data Scientist, ML Engineer, or AI Engineer in this group? I think so but I wanted to make sure I understood the intent here.